### PR TITLE
fix: [sc-20759] fix parsing  curly brackets

### PR DESCRIPTION
**Pull request by dswistowski**

### CODE REVIEW

File: tests/providers/prometheus/test_validator.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+def test_can_validate_query():
+    query = """histogram_quantile(0.95, sum by(le, namespace) (rate(httpmetrics_handled_requests_histogram_bucket{type="http", service="rpcproxy", namespace!~"rpcproxy|rpc-.*"}[1m]))) * on(namespace) group_left(eth_network) (1*kube_namespace_labels{eth_network="testnet"})"""  # noqa: E501
+    QueryValidator().validate_query(query)
+
+
+def test_can_find_closing():
+    expression = 'histogram_quantile(0.95, sum by(le, namespace) (rate(httpmetrics_handled_requests_histogram_bucket{type="http", service="rpcproxy", namespace!~"rpcproxy|rpc-.*"}[1m]))) * on(namespace) group_left(eth_network) (1*kube_namespace_labels{eth_network="testnet"})'  # noqa: E501
+    assert 167 == QueryValidator.find_closing("(", ")", expression)
+    assert expression[168:171] == " * "
```
Comment (nimbol) [inline tests/providers/prometheus/test_validator.py:12]: Not sure what this final assert statement is for.

**Review by nimbol — APPROVED** [review]
The function `extract_metric_names` now returns the following field names:
```
[
    'httpmetrics_handled_requests_histogram_bucket',
    'le',
    'namespace',
    'eth_network',
]
```

Which is still wrong, but good enough I suppose. Especially since we don't seem to use this output for anything other than validation.

I might be nice to formalise this in a test at least.

File: tests/providers/prometheus/test_validator.py (hunk 1 above)
Comment (dswistowski) [reply tests/providers/prometheus/test_validator.py:12]: to locate 168 character in the string - for this case we have ` * `... after `histogram_quantile` call

**Comment by dswistowski** [comment]
> The function `extract_metric_names` now returns the following field names:
> 
> ```
> [
>     'httpmetrics_handled_requests_histogram_bucket',
>     'le',
>     'namespace',
>     'eth_network',
> ]
> ```
> 
> Which is still wrong, but good enough I suppose. Especially since we don't seem to use this output for anything other than validation.
> 
> I might be nice to formalise this in a test at least.

This is bug fix change, was trying to make the scope minimal

**Pull request merged**